### PR TITLE
Add testing of legacy Cypher HTTP endpoint

### PR DESCRIPTION
**Pull request by Mats-SX**
This works very much like the transactional REST/HTTP endpoint, so it inherits
and uses most of the pre-existing functionality, with some minor alterations in
the JSON input and output.

### CODE REVIEW

File: enterprise/server-enterprise/src/test/java/org/neo4j/server/rest/security/CypherRESTInteraction.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        return "db/data/cypher";
+    }
+
+    CypherRESTInteraction( Map<String,String> config ) throws IOException
+    {
+        super( config );
+    }
+
+    @Override
+    public String executeQuery( RESTSubject subject, String call, Map<String,Object> params,
```
Comment (OliviaYtterbrink) [inline enterprise/server-enterprise/src/test/java/org/neo4j/server/rest/security/CypherRESTInteraction.java]: Could rewrite this in the superclass and only extract the minor parts that are different.

File: enterprise/server-enterprise/src/test/java/org/neo4j/server/rest/security/CypherRESTAuthScenariosInteractionTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        IS_EMBEDDED = false;
+    }
+
+    @Override
+    protected NeoInteractionLevel<RESTSubject> setUpNeoServer( Map<String,String> config ) throws Throwable
+    {
+        return new CypherRESTInteraction( config );
+    }
+
+    // TODO: This is removed in an upcoming PR, so it doesn't need to be fixed
```
Comment (OliviaYtterbrink) [inline enterprise/server-enterprise/src/test/java/org/neo4j/server/rest/security/CypherRESTAuthScenariosInteractionTest.java]: Needs rebase due to merging of https://github.com/neo4j/neo4j/pull/8114

**Review by OliviaYtterbrink — CHANGES_REQUESTED** [review]

File: enterprise/server-enterprise/src/test/java/org/neo4j/server/rest/security/CypherRESTAuthScenariosInteractionTest.java (hunk 2 above)
Comment (Mats-SX) [reply enterprise/server-enterprise/src/test/java/org/neo4j/server/rest/security/CypherRESTAuthScenariosInteractionTest.java]: Rebased and removed.

File: enterprise/server-enterprise/src/test/java/org/neo4j/server/rest/security/CypherRESTInteraction.java (hunk 1 above)
Comment (Mats-SX) [reply enterprise/server-enterprise/src/test/java/org/neo4j/server/rest/security/CypherRESTInteraction.java]: I did that, and also extracted all common code to an abstract super-class.

**Review by OliviaYtterbrink — APPROVED** [review]

**Pull request merged**